packages/contracts/test/voting/token-voting.ts

```diff
@@ -637,90 +657,83 @@ describe('TokenVoting', function () {
           voting.connect(signers[2]).vote(id, VoteOption.Abstain, false)
         )
           .to.emit(voting, VOTING_EVENTS.VOTE_CAST)
-          .withArgs(id, signers[2].address, VoteOption.Abstain, 1);
+          .withArgs(id, signers[2].address, VoteOption.Abstain, 10);
 
         proposal = await voting.getProposal(id);
-        expect(proposal.tally.yes).to.equal(1);
-        expect(proposal.tally.no).to.equal(1);
-        expect(proposal.tally.abstain).to.equal(1);
+        expect(proposal.tally.yes).to.equal(10);
+        expect(proposal.tally.no).to.equal(10);
+        expect(proposal.tally.abstain).to.equal(10);
       });
 
       it('reverts on vote replacement', async () => {
         await advanceIntoVoteTime(startDate, endDate);
 
-        await governanceErc20Mock.mock.getPastVotes.returns(1);
-
         await voting.vote(id, VoteOption.Yes, false);
 
         // Try to replace the vote
-        await expect(voting.vote(id, VoteOption.Yes, false)).to.be.revertedWith(
-          customError('VoteCastForbidden', id, signers[0].address)
-        );
-        await expect(voting.vote(id, VoteOption.No, false)).to.be.revertedWith(
-          customError('VoteCastForbidden', id, signers[0].address)
-        );
-        await expect(
-          voting.vote(id, VoteOption.Abstain, false)
-        ).to.be.revertedWith(
-          customError('VoteCastForbidden', id, signers[0].address)
-        );
-        await expect(
-          voting.vote(id, VoteOption.None, false)
-        ).to.be.revertedWith(
-          customError('VoteCastForbidden', id, signers[0].address)
-        );
+        await expect(voting.vote(id, VoteOption.Yes, false))
+          .to.be.revertedWithCustomError(voting, 'VoteCastForbidden')
+          .withArgs(id, signers[0].address);
+        await expect(voting.vote(id, VoteOption.No, false))
+          .to.be.revertedWithCustomError(voting, 'VoteCastForbidden')
+          .withArgs(id, signers[0].address);
+        await expect(voting.vote(id, VoteOption.Abstain, false))
+          .to.be.revertedWithCustomError(voting, 'VoteCastForbidden')
+          .withArgs(id, signers[0].address);
+        await expect(voting.vote(id, VoteOption.None, false))
+          .to.be.revertedWithCustomError(voting, 'VoteCastForbidden')
+          .withArgs(id, signers[0].address);
       });
 
       it('can execute early if participation is large enough', async () => {
         await advanceIntoVoteTime(startDate, endDate);
 
-        await governanceErc20Mock.mock.getPastVotes.returns(50);
-        await voting.vote(id, VoteOption.Yes, false);
+        await voting.connect(signers[0]).vote(id, VoteOption.Yes, false);
+        await voting.connect(signers[1]).vote(id, VoteOption.Yes, false);
+        await voting.connect(signers[2]).vote(id, VoteOption.Yes, false);
+        await voting.connect(signers[3]).vote(id, VoteOption.Yes, false);
+        await voting.connect(signers[4]).vote(id, VoteOption.Yes, false);
```

Comment:
could also be optimized with a loop and `Promise.all`

Reply:
Improved in e66e59747b75c6ce535bde30be1342d74a358978